Objective-C/Objective-C.sublime-syntax

```diff
@@ -13,200 +15,328 @@ first_line_match: |-
+
+  block-fn:
+    - meta_scope: meta.function.block.objc
+    - match: ''
+      set: [block-fn.body, block-fn.params]
+  block-fn.params:
+    - match: \(
+      scope: punctuation.section.group.begin.objc
+      set: function.params
+    - include: else-pop
+  block-fn.body:
+    - match: \{
+      scope: punctuation.section.block.begin.objc
+      set: enclosure.block
+    - match: (?=\S)
+      fail: block-fn
 
-  protocol_type_qualifier:
-    - match: \b(in|out|inout|oneway|bycopy|byref)\b
-      scope: storage.modifier.protocol.objc
+  preprocessor:
+    - meta_prepend: true
+    - match: '{{dir_start}}(import)\b'
+      scope: keyword.control.import.objc
+      push: preprocessor.include
 
-  ns-predicate:
-    - match: '(?=\])'
-      pop: true
-    - match: \bargument(?:Array|s)(:)
-      scope: support.function.any-method.name-of-parameter.objc
-      captures:
-        1: punctuation.separator.arguments.objc
-    - match: \b\w+(:)
-      scope: invalid.illegal.unknown-method.objc
-      captures:
-        1: punctuation.separator.arguments.objc
-    - match: '@"'
+  string:
+    - meta_prepend: true
+    - match: \@\"
       scope: punctuation.definition.string.begin.objc
-      push:
-        - meta_scope: string.quoted.double.objc
-        - match: '"'
-          scope: punctuation.definition.string.end.objc
-          pop: true
-        - match: \b(?i:AND|OR|NOT|IN)\b
-          scope: keyword.operator.logical.predicate.cocoa.objc
-        - match: \b(?i:ALL|ANY|SOME|NONE)\b
-          scope: constant.language.predicate.cocoa.objc
-        - match: \b(?i:NULL|NIL|SELF|TRUE|YES|FALSE|NO|FIRST|LAST|SIZE)\b
-          scope: constant.language.predicate.cocoa.objc
-        - match: \b(?i:MATCHES|CONTAINS|BEGINSWITH|ENDSWITH|BETWEEN|LIKE)\b
-          scope: keyword.operator.comparison.predicate.cocoa.objc
-        - match: \b(?i:C(ASEINSENSITIVE|I))\b
-          scope: keyword.other.modifier.predicate.cocoa.objc
-        - match: \b(?i:ANYKEY|SUBQUERY|CAST|TRUEPREDICATE|FALSEPREDICATE)\b
-          scope: keyword.other.predicate.cocoa.objc
-        - match: '\\(\\|[abefnrtv''"?]|[0-3]\d{,2}|[4-7]\d?|x[a-zA-Z0-9]+)'
-          scope: constant.character.escape.objc
-        - match: \\.
-          scope: invalid.illegal.unknown-escape.objc
-
-  unique-strings:
-    - match: '@"'
+      push: string.double-ns
+    - match: \@\'
       scope: punctuation.definition.string.begin.objc
-      push:
-        - meta_scope: string.quoted.double.objc
-        - match: '"'
-          scope: punctuation.definition.string.end.objc
-          pop: true
-        - include: scope:source.c#string_escaped_char
-        - match: '%@'
-          scope: constant.other.placeholder.objc
-        - include: scope:source.c#string_placeholder
+      push: string.single
+  string.double-ns:
+    - meta_scope: string.quoted.double.objc
+    - include: string.double
+    - match: '%@'
+      scope: constant.other.placeholder.objc
+
+  available:
```

Comment:
1. Missing meta content scope.
2. Maybe move `\)` pattern to beginning of context for max prio.

```suggestion
  available.body:
    - meta_scope: meta.available.objc  # or meta.directive.available ???
    - meta_content_scope: meta.group.objc
    - match: \)
      scope: meta.group.objc punctuation.section.group.end.objc
      pop: 1
```

Note: using `meta_content_scope` to help with properly scoping only opening \( in `- match: '(@available)\s*(\()'`